src/Command/UnregistrerAdherentsCommand.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+namespace AppBundle\Command;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Membership\AdherentManager;
+use AppBundle\Repository\AdherentRepository;
+use Doctrine\ORM\EntityManager;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Filesystem\Exception\FileNotFoundException;
+
+class UnregistrerAdherentsCommand extends ContainerAwareCommand
+{
+    const COMMAND_NAME = 'app:adherent:unregister';
+
+    /**
+     * @var EntityManager
+     */
+    private $em;
+
+    /**
+     * @var AdherentRepository
+     */
+    private $adherentRepository;
+
+    /**
+     * @var AdherentManager
+     */
+    private $adherentManager;
+
+    private $emailNotFound;
+
+    private $hosts;
+
+    private $citizenProjectAdministrators;
+
+    protected function configure()
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->addArgument('fileUrl', InputArgument::REQUIRED)
+            ->setDescription('Unregistrate adherents from CSV file (only the fifth column "email" is taken into account)');
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        $this->em = $this->getContainer()->get('doctrine.orm.entity_manager');
+        $this->adherentRepository = $this->em->getRepository(Adherent::class);
+        $this->adherentManager = $this->getContainer()->get('app.adherent_manager');
+        $this->emailNotFound = [];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        try {
+            $emails = $this->parseCSV($input->getArgument('fileUrl'), 'r');
+        } catch (FileNotFoundException $exception) {
+            $output->writeln(
+                sprintf('%s file not found', $input->getArgument('fileUrl'))
+            );
+
+            return 1;
+        }
+
+        $this->em->beginTransaction();
+
+        if ($this->unregistrateAdherents($emails, $output)) {
+            return 1;
+        }
+
+        $this->em->commit();
+
+        $output->writeln('Unregistration finished.');
+
+        if (0 < count($this->emailNotFound)) {
```

Comment:
`if ($this->emailNotFound) {`, same below.

Reply:
Done